### PR TITLE
PR: Setup Amazon S3

### PR DESCRIPTION
**Pull request by ababber**
### What this PR does (required):
- setup amazon s3 w/ .env variables, created upload route, tested route and upload to amazon s3 with postman

### Screenshots / Videos (front-end only):
Screen Shot 2021-11-23 at 10 44 40 (2)

### Any information needed to test this feature (required):
- use iam user credentials for amazon s3 to setup .env variables, setup postman for route test, use images from local disk to upload to amazon s3 using postman

### Any issues with the current functionality (optional):
- the profile model is not created in db upon user creation, there is an issue in server/controllers/auth.js that results in profile id in user model to be undefined, once the profile model is setup correctly, image urls can be easily written to it

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
Looks good!

File: server/controllers/upload.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const uploadToS3 = (img) => {
+  const params = {
+    Bucket: BUCKET_NAME,
+    Key: img.originalname,
+    Body: img.buffer,
+  }
+  return s3.upload(params).promise();
+};
+
+exports.multerMultiUpload = multer({storage: multer.memoryStorage()}).array('images');
```
Comment (rajivtitus) [inline server/controllers/upload.js:23]: Optional: You can also add a check to accept only image file types (jpeg/png)

**Review by rajivtitus — APPROVED** [review]
Good work 😄

File: server/controllers/upload.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  const { _id } = req.body;
+
+  for(let i = 0; i < images.length; i++){
+    let img = images[i];
+    promises.push(uploadToS3(img));
+  }
+
+  try {
+    const data = await Promise.all(promises)
+    try {
```
Comment (rajivtitus) [inline server/controllers/upload.js:47]: You don't need two try/catch block here, one should suffice

File: server/controllers/upload.js (hunk 1 above)
Comment (ababber) [reply server/controllers/upload.js]: I implemented it, check out the latest commit!

**Pull request merged**